pmd-core/src/main/java/net/sourceforge/pmd/renderers/RendererFactory.java

```diff
@@ -12,155 +9,90 @@
 import java.util.Properties;
 import java.util.Set;
 import java.util.TreeMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import net.sourceforge.pmd.properties.PropertyDescriptor;
 import net.sourceforge.pmd.util.AssertionUtil;
 
-/**
- * This class handles the creation of Renderers.
- *
- * @see Renderer
- */
 public final class RendererFactory {
 
     private static final Logger LOG = LoggerFactory.getLogger(RendererFactory.class);
 
-    private static final Map<String, Class<? extends Renderer>> REPORT_FORMAT_TO_RENDERER;
+    private static final Map<String, Function<Properties, Renderer>> RENDERERS;
 
     static {
-        Map<String, Class<? extends Renderer>> map = new TreeMap<>();
-        map.put(CodeClimateRenderer.NAME, CodeClimateRenderer.class);
-        map.put(XMLRenderer.NAME, XMLRenderer.class);
-        map.put(IDEAJRenderer.NAME, IDEAJRenderer.class);
-        map.put(TextColorRenderer.NAME, TextColorRenderer.class);
-        map.put(TextRenderer.NAME, TextRenderer.class);
-        map.put(TextPadRenderer.NAME, TextPadRenderer.class);
-        map.put(EmacsRenderer.NAME, EmacsRenderer.class);
-        map.put(CSVRenderer.NAME, CSVRenderer.class);
-        map.put(HTMLRenderer.NAME, HTMLRenderer.class);
-        map.put(XSLTRenderer.NAME, XSLTRenderer.class);
-        map.put(YAHTMLRenderer.NAME, YAHTMLRenderer.class);
-        map.put(SummaryHTMLRenderer.NAME, SummaryHTMLRenderer.class);
-        map.put(VBHTMLRenderer.NAME, VBHTMLRenderer.class);
-        map.put(EmptyRenderer.NAME, EmptyRenderer.class);
-        map.put(JsonRenderer.NAME, JsonRenderer.class);
-        map.put(SarifRenderer.NAME, SarifRenderer.class);
-        REPORT_FORMAT_TO_RENDERER = Collections.unmodifiableMap(map);
+        RENDERERS = Map.of(
+            CodeClimateRenderer.NAME, CodeClimateRenderer::new,
+            XMLRenderer.NAME, XMLRenderer::new,
```

Comment:
`XMLRenderer` class has two constructors, it cannot use this way to init. please switch this back with the parameter invocation.